050-AIS-APIManagementwithFunctions/Student/Challenge-04.md

```diff
@@ -0,0 +1,73 @@
+# Challenge 04 - Securing Backends APIs
+
+
+[<Previous Challenge](./Challenge-03.md) - **[Home](../README.md)**
+
+## Pre-requisites
+
+- You should have completed Challenge 03
+
+## Introduction
+You would like to be able to secure your backend APIs in one of the two ways:
+- Secure Hello API in a private network
+- Test end-to-end authorization to Hello API via OAuth.  
+
+## Description
+You should be able to either:
+- For Scenario 01: Configure secured backend APIs in a private network
+    - Create a new Function App in Elastic Premium plan which will be imported to APIM as Hello Internal API.  
+    - The existing API - Hello API - will now become the public/external API.  The new path should configured in APIM as: https://apim-{{unique_id}}.azure-api.net/external/hello
+    - Secure internal Hello Function App by enabling networking feature by either:
+        - Only accepts traffic coming from the APIM subnet
+        - Assigning a private endpoint to the Function App
+    - Import the new Function App as Hello Internal API to APIM.  The new path should be: https://apim-{{unique_id}}.azure-api.net/internal/hello
+    - Secure external Hello API so that it would only accept requests routed from Application Gateway, which includes setting-up an APIM policy.
+    - To allow routes to external Hello API only, you should configure URL redirection mechanism in Application Gateway so that:
+        - All calls to the AGW endpoint with the path /external/* (http://pip-{{unique_id}}.australiaeast.cloudapp.azure.com/external)  would go to https://api.{{unique_id}}.azure-api.net/external/hello
+        - While calls to the default path http://pip-{{unique_id}}.australiaeast.cloudapp.azure.com/ returns HTTP 404.
+
+- For Scenario 02: Configure OAuth2 authorization when calling Hello API
```

Comment:
Make this a sub-header with 3 hashes.

"### Scenario 2: Configure OAuth2 authorization when calling into Hello API"